source/common/http/alternate_protocols_cache_impl.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "envoy/common/http/alternate_protocols_cache.h"
+#include "envoy/common/optref.h"
+#include "envoy/common/time.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Http {
+
+// Tracks alternate protocols that can be used to make an HTTP connection to an origin server.
+// See https://tools.ietf.org/html/rfc7838 for HTTP Alternate Services and
+// https://datatracker.ietf.org/doc/html/draft-ietf-dnsop-svcb-https-04 for the
+// "HTTPS" DNS resource record.
```

Comment:
update class comments for less copy/pasting?
an implementation of AlternateProtocolsCache [anything special or unique]

Reply:
Done. (Will expand the comment in the larger PR when it does more, like use TLS)